src/libponyrt/sched/scheduler.c

```diff
@@ -333,16 +388,63 @@ static pony_actor_t* steal(scheduler_t* sched)
     //    stealing for generating far fewer block/unblock messages.
     if (!block_sent)
     {
-      if (steal_attempts < scheduler_count)
+      uint32_t current_active_scheduler_count = get_active_scheduler_count();
+      if (steal_attempts < current_active_scheduler_count)
       {
         steal_attempts++;
       }
-      else if ((!sched->asio_noisy) &&
-        ((tsc2 - tsc) > 1000000) &&
+      else if (((tsc2 - tsc) > 1000000) &&
         (ponyint_mutemap_size(&sched->mute_mapping) == 0))
       {
-        send_msg(0, SCHED_BLOCK, 0);
-        block_sent = true;
+        // if we're the highest active scheduler thread
+        // and there are more active schedulers than the minimum requested
+        if ((sched == &scheduler[current_active_scheduler_count - 1])
+          && (current_active_scheduler_count > min_scheduler_count) &&
+          !atomic_exchange_explicit(&scheduler_count_changing, true,
+            memory_order_acquire))
+        {
+          // let sched 0 know we're suspending
+          send_msg(0, SCHED_SUSPEND, 0);
+
+          // decrement active_scheduler_count so other schedulers know we're
+          // sleeping
+          uint32_t sched_count = atomic_load_explicit(&active_scheduler_count,
+            memory_order_relaxed);
+          atomic_store_explicit(&active_scheduler_count, sched_count - 1,
+            memory_order_relaxed);
+
+          // unlock the bool that controls modifying the active scheduler count
+          // variable
+          atomic_store_explicit(&scheduler_count_changing, false,
+            memory_order_relaxed);
```

Comment:
`memory_order_release` here since the operation marks the end of a critical section.